crates/services/consensus_module/poa/src/pre_confirmation_signature_service/tests.rs

```diff
@@ -0,0 +1,395 @@
+#![allow(non_snake_case)]
+
+use super::*;
+use crate::pre_confirmation_signature_service::error::Error;
+use fuel_core_services::StateWatcher;
+use fuel_core_types::fuel_tx::Transaction;
+use std::{
+    sync::Arc,
+    time::Duration,
+};
+use tokio::sync::Notify;
+
+use fuel_core_types::fuel_types::BlockHeight;
+
+#[allow(dead_code)]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum Status {
+    Success { height: BlockHeight },
+    Fail { height: BlockHeight },
+    SqueezedOut { reason: String },
+}
+
+struct FakeTxReceiver {
+    recv: tokio::sync::mpsc::Receiver<Vec<(Transaction, Status)>>,
+}
+
+impl TxReceiver for FakeTxReceiver {
+    type Txs = Vec<(Transaction, Status)>;
+
+    async fn receive(&mut self) -> Result<Vec<(Transaction, Status)>> {
+        let txs = self.recv.recv().await;
+        match txs {
+            Some(txs) => Ok(txs),
+            None => Err(Error::TxReceiverError("No txs received".into())),
+        }
+    }
+}
+
+struct FakeBroadcast {
+    delegation_key_sender: tokio::sync::mpsc::Sender<FakeSignedData<FakeSigningKey>>,
+    tx_sender: tokio::sync::mpsc::Sender<FakeSignedData<Vec<(Transaction, Status)>>>,
+}
+
+impl Broadcast for FakeBroadcast {
+    type PreConfirmations = Vec<(Transaction, Status)>;
+    type ParentSignature<T> = FakeSignedData<T>;
+    type DelegateKey = FakeSigningKey;
+
+    async fn broadcast_txs(
+        &mut self,
+        txs: <Self::DelegateKey as SigningKey>::Signature<Self::PreConfirmations>,
```

Comment:
you could make this type a little simpler

```
type Signed<K, T> = <K as SigningKey>::Signature<T>;
```

and then

```suggestion
        txs: Signed<Self::DelegateKey, Self::Preconfirmations>
```